tests/TestCase/Routing/Route/PluginShortRouteTest.php

```diff
@@ -57,7 +57,7 @@ public function testParsing(): void
      */
     public function testMatch(): void
     {
-        $route = new PluginShortRoute('/{plugin}', ['action' => 'index'], ['plugin' => 'foo|bar']);
+        $route = new PluginShortRoute('/{plugin}', ['action' => 'index'], ['plugin' => 'Foo|Bar']);
```

Comment:
So the drawback of not inflecting the URL parts before matching is that the regexes for matching the parts would have be written considering the casing used in URL array rather than the URL string. This could cause a lot of useland breakage, so I don't think this change is a good idea.

Reply:
Should the test setup on line 44 also change? This looks like it would only parse `/Foo` or `/Bar` instead of `/foo` and `/bar`.